### DISCUSSION TITLE
NVMe cli

### DISCUSSION THREAD

**Discussion by forgosh** [siderolabs/talos#9879] [Q&A]
It appears that nvme-cli is not included in the base image. Is there an extension to add that?

**Comment by smira** [comment]
What is the reason you're looking for `nvme-cli` on the host filesystem?

Please keep in mind, that you can use Kubernetes privileged pods and do any operation from it (bringing your own tools).

**Comment by forgosh** [comment]
I’m looking at nvme-cli in order to use NetApp Trident to provide PVs to pods.

Seth Forgosh
Manager Hybrid Enterprise Apps Solution Architects
908.415.0911

Need to book a meeting? Check my schedule ***@***.***/fa8a07623e56497a916a308f16eaa7316147967676235001718/calendar.html>.
[A picture containing text  Description automatically generated]

[signature_1749753811]<https://www.facebook.com/netapp>    [signature_4093644817] <https://www.instagram.com/netapp>      [signature_4222464626] <https://www.youtube.com/netapp>      [signature_931277118] <https://www.linkedin.com/company/netapp>      [signature_2450387534] <https://www.twitter.com/NetApp>


From: Andrey Smirnov ***@***.***>
Date: Thursday, December 5, 2024 at 5:52 AM
To: siderolabs/talos ***@***.***>
Cc: Forgosh, Seth ***@***.***>, Author ***@***.***>
Subject: Re: [siderolabs/talos] NVMe cli (Discussion #9879)
EXTERNAL EMAIL - USE CAUTION when clicking links or attachments



What is the reason you're looking for nvme-cli on the host filesystem?

Please keep in mind, that you can use Kubernetes privileged pods and do any operation from it (bringing your own tools).

**Comment by forgosh** [comment]
NVMe is included in the base image however the client for using NVMe/TCP is not. That is the piece that I need to get installed.

Seth Forgosh
Manager Hybrid Enterprise Apps Solution Architects
908.415.0911

Need to book a meeting? Check my schedule ***@***.***/fa8a07623e56497a916a308f16eaa7316147967676235001718/calendar.html>.
[A picture containing text  Description automatically generated]

[signature_1749753811]<https://www.facebook.com/netapp>    [signature_4093644817] <https://www.instagram.com/netapp>      [signature_4222464626] <https://www.youtube.com/netapp>      [signature_931277118] <https://www.linkedin.com/company/netapp>      [signature_2450387534] <https://www.twitter.com/NetApp>


From: Forgosh, Seth ***@***.***>
Date: Thursday, December 5, 2024 at 7:39 AM
To: siderolabs/talos ***@***.***>, siderolabs/talos ***@***.***>
Cc: Author ***@***.***>
Subject: Re: [siderolabs/talos] NVMe cli (Discussion #9879)
I’m looking at nvme-cli in order to use NetApp Trident to provide PVs to pods.

Seth Forgosh
Manager Hybrid Enterprise Apps Solution Architects
908.415.0911

Need to book a meeting? Check my schedule ***@***.***/fa8a07623e56497a916a308f16eaa7316147967676235001718/calendar.html>.
[A picture containing text  Description automatically generated]

[signature_1749753811]<https://www.facebook.com/netapp>    [signature_4093644817] <https://www.instagram.com/netapp>      [signature_4222464626] <https://www.youtube.com/netapp>      [signature_931277118] <https://www.linkedin.com/company/netapp>      [signature_2450387534] <https://www.twitter.com/NetApp>


From: Andrey Smirnov ***@***.***>
Date: Thursday, December 5, 2024 at 5:52 AM
To: siderolabs/talos ***@***.***>
Cc: Forgosh, Seth ***@***.***>, Author ***@***.***>
Subject: Re: [siderolabs/talos] NVMe cli (Discussion #9879)
EXTERNAL EMAIL - USE CAUTION when clicking links or attachments


What is the reason you're looking for nvme-cli on the host filesystem?

Please keep in mind, that you can use Kubernetes privileged pods and do any operation from it (bringing your own tools).

**Reply by smira** [reply]
It's not available on Talos host, but you can look into your PV provisioner to understand why it tries to find it on the host, can you make it use it something from the container, etc.

**Comment by forgosh** [comment]
How would a provisioner that is using NVMe//TCP use something else to provision a NVMe/TCP PV?

Seth Forgosh
Manager Hybrid Enterprise Apps Solution Architects
908.415.0911

Need to book a meeting? Check my schedule ***@***.***/fa8a07623e56497a916a308f16eaa7316147967676235001718/calendar.html>.
[A picture containing text  Description automatically generated]

[signature_1749753811]<https://www.facebook.com/netapp>    [signature_4093644817] <https://www.instagram.com/netapp>      [signature_4222464626] <https://www.youtube.com/netapp>      [signature_931277118] <https://www.linkedin.com/company/netapp>      [signature_2450387534] <https://www.twitter.com/NetApp>


From: Andrey Smirnov ***@***.***>
Date: Thursday, December 5, 2024 at 8:39 AM
To: siderolabs/talos ***@***.***>
Cc: Forgosh, Seth ***@***.***>, Author ***@***.***>
Subject: Re: [siderolabs/talos] NVMe cli (Discussion #9879)
EXTERNAL EMAIL - USE CAUTION when clicking links or attachments



It's not available on Talos host, but you can look into your PV provisioner to understand why it tries to find it on the host, can you make it use it something from the container, etc.

**Comment by steverfrancis** [comment]
I think Andrey is suggesting you use nvme-cli from a pod, not from the host.
